projects/self_feeding/self_feeding_agent.py

```diff
@@ -477,8 +478,20 @@ def dialog_step(self, batch):
             cands, cand_vecs, label_inds = self._build_candidates(
                 batch, source=self.opt['eval_candidates'], mode='eval'
             )
-
-        scores = self.model.score_dialog(batch.text_vec, cand_vecs)
+            if self.encode_candidate_vecs and self.eval_candidates in [
+                'fixed',
+                'vocab',
+            ]:
+                # if we cached candidate encodings for a fixed list of candidates,
+                # pass those into the score_candidates function
+                if self.eval_candidates == 'fixed':
+                    cand_encs = self.fixed_candidate_encs
+                elif self.eval_candidates == 'vocab':
+                    cand_encs = self.vocab_candidate_encs
+
+        scores = self.model.score_dialog(
+            batch.text_vec, cand_encs if cand_encs is not None else cand_vecs
```

Comment:
Why make this new `cand_encs` instead of replacing `cand_vecs`?

Reply:
ahh good point i'll change that